### PR TITLE
Rename solana-runtime to sealevel

### LINKED ISSUE #5486: No component to share with other layer 1s
# Sealevel: The Portable, Parallel Smart Contracts Layer

#### Problem

We've recently started speaking about the versatility of Solana tech we call [Sealevel](https://medium.com/solana-labs/sealevel-parallel-processing-thousands-of-smart-contracts-d814b378192) and that if other layer 1 solutions incorporated it, blockchain developers could create _portable_ smart contracts that can be executed in parallel. One problem: there's no component called Sealevel in the codebase. While everything described by that article exists, there's no shrink-wrapped package for other Layer 1 solutions to consume. The closest piece is the `solana-runtime` crate. But what about the pieces that actually do pipelining optimizations? That is, the TPU (and on the validator side, the TVU)?

#### Proposed Solution

Below the TPU, should be a language agnostic runtime. It should be responsible for safe, fast, and deterministic execution of third-party programs.  Call it "Sealevel" and make every effort to make it easily shared across layer 1 solutions and perhaps even outside blockchain. Initially, it can be a simple rename of `solana-runtime` to `sealevel`, and `solana-sdk` to `sealevel-sdk` (and a new README explaining how it's a standalone component). Once the API is stabilized, we should move it to a new git repo, `solana-labs/sealevel`.

### PR DESCRIPTION
**Pull request by garious**
#### Problem

Solana's runtime development is not benefiting from collaboration with other layer 1 solutions. Currently, it would be as awkward for another project to embed solana_runtime as it was for Solana to embed Libra's Move (fork and hack it). It's technically possible, but we don't want to discourage collaboration or encourage forks.

#### Summary of Changes

Rename solana_runtime to sealevel. This PR doesn't also rename solana_sdk, as described in the issue. Looks like that crate should first be split into two (one for programs, one for program tests).

This can go in as-is, but there's still a lot more work before this is suitable for embedding in other layer 1s. We may, for example, want to decouple sealevel from the Vote and Storage programs.

Fixes #5486

### CODE REVIEW

**Comment by mvines** [comment]
The problem description doesn't describe a problem

**Comment by garious** [comment]
better?

File: runtime/src/lib.rs [hunk 1]
```diff
@@ -1,39 +1,35 @@
...
+pub use sealevel::loader_utils;
+pub use sealevel::message_processor;
+pub use sealevel::rent_collector;
+pub use sealevel::stakes;
+pub use sealevel::status_cache;
+pub use sealevel::storage_utils;
+pub use sealevel::transaction_batch;
+pub use sealevel::transaction_utils;
+//
+//#[macro_use]
```
Comment (CriesofCarrots) [inline runtime/src/lib.rs]: Remove?

Comment (garious) [reply runtime/src/lib.rs]: Done

**Pull request merged**